### ISSUE TITLE
Building the Dockercontainer fails

### ISSUE THREAD

**Issue by Krumbelfix** [nightscout/cgm-remote-monitor#7291]
**Describe the bug**
I wanted to build the Image by myself. it failed

**To Reproduce**
1. save the following dockerfile: https://github.com/nightscout/cgm-remote-monitor/blob/master/Dockerfile
2. docker build . -t nightscout:local
3. Log:
```
root@homeserver:/home/snippedUser/docker-repo/diabetes# docker build . -t nightscout:local
Sending build context to Docker daemon  4.096kB
Step 1/8 : FROM node:14.15.3-alpine
 ---> 51d926a5599d
Step 2/8 : LABEL maintainer="Nightscout Contributors"
 ---> Using cache
 ---> f7877cfb8fd8
Step 3/8 : WORKDIR /opt/app
 ---> Using cache
 ---> 31c246a8f10d
Step 4/8 : ADD . /opt/app
 ---> Using cache
 ---> ad609391a3e2
Step 5/8 : RUN npm install --cache /tmp/empty-cache &&   npm run postinstall &&   npm run env &&   rm -rf /tmp/*
 ---> Running in 45a0764d9724
npm WARN saveError ENOENT: no such file or directory, open '/opt/app/package.json'
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN enoent ENOENT: no such file or directory, open '/opt/app/package.json'
npm WARN app No description
npm WARN app No repository field.
npm WARN app No README data
npm WARN app No license field.

up to date in 0.491s
found 0 vulnerabilities

npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /opt/app/package.json
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, open '/opt/app/package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-01-10T19_20_52_611Z-debug.log
The command '/bin/sh -c npm install --cache /tmp/empty-cache &&   npm run postinstall &&   npm run env &&   rm -rf /tmp/*' returned a non-zero code: 254
```

**Expected behavior**
images gets build properly

**Your setup information**
- Ubuntu 20.04
- docker-compose 1.25.0-1
- docker.io 20.10.7-0ubuntu5~20.04.2
- python3-docker 4.1.0-1
- python3-dockerpty 0.4.1-2

**Comment by Krumbelfix** [comment]
After modifying the Dockerfile it still fails:
**Dockerfile**:

```
FROM node:14.15.3-alpine

LABEL maintainer="Nightscout Contributors"

RUN mkdir -p /opt/app
COPY ./package.json /opt/app/package.json
WORKDIR /opt/app
ADD . /opt/app

# TODO: We should be able to do `RUN npm install --only=production`.
# For this to work, we need to copy only package.json and things needed for `npm`'s to succeed.
# TODO: Do we need to re-add `npm audit fix`? Or should that be part of a development process/stage?
RUN npm install --cache /tmp/empty-cache && \
  npm run postinstall && \
  npm run env && \
  rm -rf /tmp/*
  # TODO: These should be added in the future to correctly cache express-minify content to disk
  # Currently, doing this breaks the browser cache.
  # mkdir /tmp/public && \
  # chown node:node /tmp/public

USER node
EXPOSE 1337

CMD ["node", "lib/server/server.js"]
```

**package.json**
https://github.com/nightscout/cgm-remote-monitor/blob/14.2.5/package.json

**Log**
```
root@homeserver:/home/snippedUser/docker-repo/diabetes# docker build . -t nightscout:local
Sending build context to Docker daemon  10.24kB
Step 1/10 : FROM node:14.15.3-alpine
 ---> 51d926a5599d
Step 2/10 : LABEL maintainer="Nightscout Contributors"
 ---> Using cache
 ---> f7877cfb8fd8
Step 3/10 : RUN mkdir -p /opt/app
 ---> Using cache
 ---> 8754e338e530
Step 4/10 : COPY ./package.json /opt/app/package.json
 ---> a5a043cf6ef9
Step 5/10 : WORKDIR /opt/app
 ---> Running in de75f95af957
Removing intermediate container de75f95af957
 ---> 9ceac50887f6
Step 6/10 : ADD . /opt/app
 ---> a6f2b7659b29
Step 7/10 : RUN npm install --cache /tmp/empty-cache &&   npm run postinstall &&   npm run env &&   rm -rf /tmp/*
 ---> Running in 1b2406233b1d
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm WARN deprecated codacy-coverage@3.4.0: Package being deprecated in favor of https://github.com/codacy/codacy-coverage-reporter
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated superagent@3.8.3: Please upgrade to v7.0.1+ of superagent.  We have fixed numerous issues with streams, form-data, attach(), filesystem errors not bubbling up (ENOENT on attach()), and all tests are now passing.  See the releases tab for more information at <https://github.com/visionmedia/superagent/releases>. Thanks to @shadowgate15, @spence-s, and @niftylettuce. Superagent is sponsored by Forward Email @ <https://forwardemail.net>
npm WARN deprecated joi@13.7.0: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated querystring@0.2.1: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated request-promise@4.2.6: request-promise has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated left-pad@1.3.0: use String.prototype.padStart()
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated axios@0.19.2: Critical security vulnerability fixed in v0.21.1. For more information, see https://github.com/axios/axios/pull/3410
npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated formidable@1.2.6: Please upgrade to latest, formidable@v2 or formidable@v3! Check these notes: https://bit.ly/2ZEqIau
npm WARN deprecated hoek@5.0.4: This version has been deprecated in accordance with the hapi support policy (hapi.im/support). Please upgrade to the latest version to get the best features, bug fixes, and security patches. If you are unable to upgrade at this time, paid support is available for older versions (hapi.im/commercial).
npm WARN deprecated topo@3.0.3: This module has moved and is now available at @hapi/topo. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
npm WARN deprecated hoek@6.1.3: This module has moved and is now available at @hapi/hoek. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
npm WARN deprecated mkdirp@0.5.1: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

> ejs@2.7.4 postinstall /opt/app/node_modules/ejs
> node ./postinstall.js

Thank you for installing EJS: built with the Jake JavaScript build tool (https://jakejs.com/)


> nodemon@1.19.4 postinstall /opt/app/node_modules/nodemon
> node bin/postinstall || exit 0

Love nodemon? You can now support the project via the open collective:
 > https://opencollective.com/nodemon/donate

npm WARN lifecycle nightscout@14.2.5~postinstall: cannot run in wd nightscout@14.2.5 webpack --mode production --config webpack/webpack.config.js && npm run-script generate-keys (wd=/opt/app)
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN notsup Unsupported engine for minimed-connect-to-nightscout@1.5.2: wanted: {"node":"^10.14.1 || ^8.15.1 || ^12.2.0","npm":"^6.4.1"} (current: {"node":"14.15.3","npm":"6.14.9"})
npm WARN notsup Not compatible with your version of node/npm: minimed-connect-to-nightscout@1.5.2
npm WARN notsup Unsupported engine for share2nightscout-bridge@0.2.8: wanted: {"node":"12.x || 10.x || 8.x"} (current: {"node":"14.15.3","npm":"6.14.9"})
npm WARN notsup Not compatible with your version of node/npm: share2nightscout-bridge@0.2.8
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.1 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7 (node_modules/nodemon/node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

added 1514 packages from 1195 contributors and audited 1293 packages in 69.162s

67 packages are looking for funding
  run `npm fund` for details

found 22 vulnerabilities (3 low, 13 moderate, 6 high)
  run `npm audit fix` to fix them, or `npm audit` for details

> nightscout@14.2.5 postinstall /opt/app
> webpack --mode production --config webpack/webpack.config.js && npm run-script generate-keys

[webpack-cli] Failed to load '/opt/app/webpack/webpack.config.js' config
[webpack-cli] Error: Cannot find module '/opt/app/webpack/webpack.config.js'
Require stack:
- /opt/app/node_modules/webpack-cli/lib/webpack-cli.js
- /opt/app/node_modules/webpack-cli/lib/bootstrap.js
- /opt/app/node_modules/webpack-cli/bin/cli.js
- /opt/app/node_modules/webpack/bin/webpack.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at WebpackCLI.tryRequireThenImport (/opt/app/node_modules/webpack-cli/lib/webpack-cli.js:244:16)
    at loadConfigByPath (/opt/app/node_modules/webpack-cli/lib/webpack-cli.js:1710:30)
    at /opt/app/node_modules/webpack-cli/lib/webpack-cli.js:1765:11
    at Array.map (<anonymous>)
    at WebpackCLI.loadConfig (/opt/app/node_modules/webpack-cli/lib/webpack-cli.js:1764:24)
    at WebpackCLI.createCompiler (/opt/app/node_modules/webpack-cli/lib/webpack-cli.js:2185:29) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/opt/app/node_modules/webpack-cli/lib/webpack-cli.js',
    '/opt/app/node_modules/webpack-cli/lib/bootstrap.js',
    '/opt/app/node_modules/webpack-cli/bin/cli.js',
    '/opt/app/node_modules/webpack/bin/webpack.js'
  ]
}
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! nightscout@14.2.5 postinstall: `webpack --mode production --config webpack/webpack.config.js && npm run-script generate-keys`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the nightscout@14.2.5 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-01-11T09_33_08_155Z-debug.log
The command '/bin/sh -c npm install --cache /tmp/empty-cache &&   npm run postinstall &&   npm run env &&   rm -rf /tmp/*' returned a non-zero code: 2
```

**Comment by Krumbelfix** [comment]
Okay, i had to use the whole repo :)
Problem solved!

git clone https://github.com/nightscout/cgm-remote-monitor.git
cd cgm-remote-monitor
docker build . -t nightscout:local
